### PR TITLE
Clean up variable and method names

### PR DESCRIPTION
**Pull request by jbduncan**
Issue: #101

### CODE REVIEW

**Comment by tomnelson** [comment]
Do you have metrics for any impact on performance?

**Comment by jbduncan** [comment]
@tomnelson Admittedly I do not. It's not really clear to me how I'd measure the performance impact of this PR. Do you have any thoughts on this?

**Comment by tomnelson** [comment]
I would make unit tests that run the affected methods with very large
input, then run those tests with and without the changes to Streams to see
if there is a difference. In any event, having the unit tests seems like it
would be a good thing.

**Comment by tomnelson** [comment]
I'm not going to weigh in about whether readability is improved, but when I run this:
`
import java.util.*;

public class Performance {

    public static void main(String [] args) {

	double [] dist = {5, 6, 7, 8, 10, 20, 80, 100, 120, 50};
	long ts = System.currentTimeMillis();
	for (int j = 0; j < 1000; j++) {
	    double total = 0;

	    for (int i = 0; i < dist.length; i++) {
		if (dist[i] > 0) {
		    total += dist[i] * Math.log(dist[i]);
		}
	    }
	}
	System.out.println("forloop took " + (System.currentTimeMillis()-ts));

	ts = System.currentTimeMillis();
	for (int j = 0; j < 1000; j++) {

	    double total =
		Arrays.stream(dist)
		.filter(element -> element > 0)
		.map(element -> element * Math.log(element))
		.sum();
	}
	System.out.println("Streams took " + (System.currentTimeMillis()-ts));
    }
}
`
I find that using stream takes about 100ms longer. The time does not grow much when you increase the size of the loop, so maybe the extra time is because of the creation of objects. 
Is extra time for the benefit of modernized source code a worthwhile pursuit in algorithm implementations?

**Comment by jbduncan** [comment]
@tomnelson Good question, I'm honestly not sure. But regardless, I think benchmarking code as you've done in your code example above should at best be taken with a grain of salt, because the JVM can do all sorts of weird optimisations that can skew the results greatly. For example, just simply switching the order in which the for-loop-based and Stream-based code examples above run could cause them to run in very different amounts of time.

The closest-to-surefire way to benchmark this is to use JMH. Without results backed by a JMH benchmark, we'd be taking nothing more than a stab in a dark. Furthermore, it's not clear to me that the effort I'd need to put into learning how to use JMH is worth it yet.

**Comment by jbduncan** [comment]
@tomnelson But having said this, I'm more than happy to revert the Stream-related changes I've made in this PR (once I have the time and energy) if they're causing you or @jrtom to feel uneasy. :)

**Comment by jbduncan** [comment]
I've just found the time recently to have another go at this!

I've decided to remove the changes I made to stream-ify the code base in favour of keeping the existing old-style looping code, partially for perceived performance reasons and also because we currently lack tests to check things still work properly.

**Comment by jbduncan** [comment]
Additional unit tests will be done at another time.

File: jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/util/DiscreteDistribution.java [hunk 1]
```diff
@@ -188,17 +191,17 @@ public static void normalize(double[] counts, double alpha) {
    * @return the mean of the distributions
    */
   public static double[] mean(double[][] distributions) {
-    double[] d_mean = new double[distributions[0].length];
-    for (int j = 0; j < d_mean.length; j++) {
-      d_mean[j] = 0;
-    }
+    // TODO: Consider checking that the array and inner arrays of `distributions` are non-empty
+    // TODO: Consider using com.google.common.math.Stats.meanOf
```
Comment (jrtom) [inline jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/util/DiscreteDistribution.java]: Feel free to go ahead and make that change here, if you like.

**Comment by jrtom** [comment]
@jbduncan welcome back.  :)

Looks reasonable.  I'm not too concerned about fixing internal variable names, but it doesn't hurt anything to make these changes.

**Pull request merged**